_posts/en/newsletters/2019-01-22-newsletter.md

```diff
@@ -0,0 +1,111 @@
+  along LN like normal using the the hash of the preimage.  When the
+  payment reaches the receiver, they decrypt the preimage and disclose
+  it to the routing nodes in order to claim the payment.
+
+    Allowing spontaneous payments allows nodes to accept donations
+    without using extra software for generating invoices.  It can also
+    help in cases where users just want to do ad hoc payment
+    tracking, for example you initiate a 10 mBTC withdrawal from an
+    exchange and either 10 mBTC shows up in your balance within a few
+    moments or you contact support.  For serious payment tracking, users
+    should still continue to generate invoices which can be uniquely
+    associated with particular orders or other expected payments.
+
+    Osuntokun's pull request for LND is still marked as a work in
+    progress as of this writing, so we don't know yet when the feature
+    will become generally available to LND users or whether other LN
+    implementations will also provide the same feature in a compatible
+    way.
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+and [libsecp256k1][libsecp256k1 repo].*
+
+- [Bitcoin Core #14941][] makes the `unloadwallet` RPC synchronous.
+  It won't return now until the specified wallet has finished being
+  unloaded.
+
+- [Bitcoin Core #14982][] adds a new `getrpcinfo` RPC that provides
+  information about the RPC interface.  Right now it returns an
+  `active_commands` array listing all RPCs that haven't returned yet.
+
+- [LND #2448][] adds a standalone watchtower, allowing it to "negotiate
+  sessions with clients, accept state updates for active sessions,
+  monitor the chain for breaches matching known breach hints, [and] publish
+  reconstructed justice transactions on behalf of tower clients."
+  This is one of the last pieces of an initial watchtower implementation
+  than can help protect LN nodes that are offline from having their
+  funds stolen---a feature that's an important part of making LN mature
+  enough for general use.
+
+- [LND #2439][] adds the default policy for the watchtower, such as
+  allowing the tower to handle a maximum of 1,024 updates from a client
+  in a single session, allowing the watchtower to receive a reward of 1%
+  of the channel capacity if the tower ends up defending the channel,
+  and setting the default onchain feerate for justice transactions
+  (breach remedy transactions).
+
+- [LND #2198][] gives the `sendcoins` RPC a new `sweepall` parameter
+  that will spend all of the wallet's bitcoins to the specified address
+  without the user having to manually specify the amount.
+
+- [C-Lightning #2232][] extends the `listpeers` command with a new
+  `funding_allocation_msat` field that returns the amounts initially
+  placed into a channel by each peer.
+
+- [C-Lightning #2234][] extends the `listchannels` RPC to take a
```

Comment:
I was going to ask whether this was the right PR since the PR title is so unspecific and lacking in notes, but you're right. Good code archeology!

Reply:
Thanks!  Actually, I almost never read the PRs, so I'm glad you check them.  In the case of C-Lightning, the code archeology always goes the other way---the trick isn't finding out what the PR does, it's finding out which PR the commit goes with!